File: rocky.js
// Function to generate computer choice
function getComputerChoice() {
  const choices = ['rock', 'paper', 'scissors'];
  return choices[Math.floor(Math.random() * 3)];
}

// Function to play a single round
function playRound(playerSelection, computerSelection) {
  // Check if player selection is valid
  if (!['rock', 'paper', 'scissors'].includes(playerSelection.toLowerCase())) {
    return "Invalid input. Please choose 'rock', 'paper', or 'scissors'.";
  }

  playerSelection = playerSelection.toLowerCase();
  if (playerSelection === computerSelection) {
    return `It's a tie! You both chose ${playerSelection}.`;
  } else if (
    (playerSelection === 'rock' && computerSelection === 'scissors') ||
    (playerSelection === 'paper' && computerSelection === 'rock') ||
    (playerSelection === 'scissors' && computerSelection === 'paper')
  ) {
    return `You win! ${playerSelection} beats ${computerSelection}.`;
  } else {
    return `You lose! ${computerSelection} beats ${playerSelection}.`;
  }
}

// Main game function
function game() {
  let playerScore = 0;
  let computerScore = 0;

  for (let i = 0; i < 5; i++) {
    const playerSelection = prompt(`Round ${i + 1} - Choose rock, paper or scissors`);
    const computerSelection = getComputerChoice();
    const roundResult = playRound(playerSelection, computerSelection);
    console.log(roundResult);

    if (roundResult.startsWith('You win')) {
      playerScore++;
    } else if (roundResult.startsWith('You lose')) {
      computerScore++;
    }
  }

  console.log('Game over!');
  if (playerScore > computerScore) {
    console.log(`You won the game! Final score: ${playerScore} - ${computerScore}`);
  } else if (playerScore < computerScore) {
    console.log(`You lost the game! Final score: ${playerScore} - ${computerScore}`);
  } else {
    console.log(`It's a tie game! Final score: ${playerScore} - ${computerScore}`);
  }
}

// Call the game function to play the game
game();

File: rocky.py
import random

choices = ['rock', 'paper', 'scissors']
player_score = 0
cpu_score = 0
rounds = 0
question = f'Round {rounds + 1} of 5 - Rock, paper or scissors? '

def get_cpu_choices():
    return choices[random.randint(0, 2)]

def play_round(player, cpu, rounds, player_score, cpu_score):
    rounds += 1
    if player == cpu:
        print(f'Tie - both chose {player}')
    elif player == 'rock' and cpu == 'scissors' or player == 'paper' and cpu == 'rock' or player == 'scissors' and cpu == 'paper':
        player_score += 1
        print(f'Player wins - {player} beats {cpu}')
    else:
        cpu_score += 1
        print(f'CPU wins - {cpu} beats {player}')
    return rounds, player_score, cpu_score

while rounds < 5:
    player_choice = input(question)
    cpu_choice = get_cpu_choices()
    rounds, player_score, cpu_score = play_round(player_choice, cpu_choice, rounds, player_score, cpu_score)
    question = f'Round {rounds + 1} of 5 - Rock, paper or scissors? '

if player_score > cpu_score:
    print(f'Player wins! {player_score} beats {cpu_score}')
elif cpu_score > player_score:
    print(f'CPU wins! {cpu_score} beats {player_score}')
else:
    print(f'Tie - both have {player_score} points')
